tools/datamart_benchmarker.py

```diff
@@ -94,24 +115,45 @@ def benchmark(filename, query):
         }
         result_row = RESULT_TEMPLATE.copy()
         result_row['datetime'] = datetime.datetime.now().isoformat()
+        result_row['server'] = server
         result_row['resource_id'] = record['resource_hdx_id']
         result_row['resource_name'] = record['resource_name']
         result_row['download_url'] = record['download_url']
         result_row['size'] = record['size']
 
         try:
-            with Client(base_url=DATAMART_ROOT_URL, params=params, timeout=60) as ac:
+            with Client(base_url=root_url, params=params, timeout=60) as ac:
                 response = ac.get('/api/v1/datamart/data')
+            result_row['status_code'] = response.status_code
             response.raise_for_status()
             query_time = f'{time.time() - t0:0.2f}'
             print(f'{query_time}, {len(response.json()['data'])}', flush=True)
             result_row['returned_row_count'] = len(response.json()['data'])
+            result_row['returned_col_count'] = len(response.json()['data'][0].keys())
+            result_row['n_unnamed_columns'] = sum(
+                [1 for x in response.json()['data'][0].keys() if 'unnamed' in x.lower()]
+            )
+
             result_row['query_time'] = query_time
             result_row['success'] = True
+            result_row['file_format'] = response.json()['resource_metadata']['format']
```

Comment:
you _might_ not receive json 100% of the time (e.g. when you hit the rate limit).

Reply:
Yes, I've wrapped this is a general try: except block which is poor form but catches everything.

I seem to get timeouts rather than 429s when I hit the rate limit, and they happen back at the initial search around line 89. It's a bit difficult to debug because I break my code for an indeterminate time when I hit the rate limit!